### PR TITLE
[WIP] Test Harness Class Intialisation

### PR DESCRIPTION
**Pull request by abhisheksaran**
Fixes convict-git#61

### CODE REVIEW

**Comment by abhisheksaran** [comment]
WIP...

**Comment by Rahul-R5** [comment]
@abhisheksaran Make the sync changes as requested in the #17 PR. Refer the github workflow in discord and add Manas as contributor to your branch/repo. Change destination branch to 74_test_harness. 
@abhisheksaran and @stalker0418, finish the test harness fast. You can't extend this till Nov 10, you need to write unit tests for each function you have written.

**Comment by Rahul-R5** [comment]
[Test Harness function runAll](https://trello.com/c/JDv6eiah/37-test-harness-function-runall)

**Comment by Rahul-R5** [comment]
[Test Harness functions runByName](https://trello.com/c/79Mn8dNC/36-test-harness-functions-runbyname)

**Comment by Rahul-R5** [comment]
[Test Harness Function RunByCategory](https://trello.com/c/VdiaIF7o/31-test-harness-function-runbycategory)

**Comment by Rahul-R5** [comment]
[Test Harness Function RunByPriority](https://trello.com/c/NY2rJZVM/30-test-harness-function-runbypriority)

**Comment by Rahul-R5** [comment]
[Verifying Assembler Functions](https://trello.com/c/y9LhlDbd/32-verifying-assembler-functions)

**Comment by abhisheksaran** [comment]
Do you want me to create another branch and delete this pull request?

File: src/infrastructure/validation/testing/TestHarness.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
+/**
+ * @author        Abhishek Saran
+ * @author        Manas Sanjay
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java]: You can put these in the same line

File: src/infrastructure/validation/testing/TestHarness.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  void runAll(){
+  
+    //get all the test cases in infrastucture module
+ 	  File testFolderInfrastucture = new File("../../../infrastucture/tests");
+	  File[] listOfTests = testFolder.listFiles();
+	  System.out.println("Running all the test cases of infrastucture module...");
+	  for (File file : listOfTests){
+    if (file.isFile()) {
+        runByName(file.getName());
+    }
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java]: Indentation

File: src/infrastructure/validation/testing/TestHarness.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+  // Manas's code finishes here
+
+
+
+  void runAll(){
+  
+    //get all the test cases in infrastucture module
+ 	  File testFolderInfrastucture = new File("../../../infrastucture/tests");
+	  File[] listOfTests = testFolder.listFiles();
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java]: You can reduce the code by looping through all module names. You have a lot of repeated code. This becomes unmanagable if more modules are introduced. You can read through the names of the folders in src/ and populate your module name list.

File: src/infrastructure/validation/testing/TestHarness.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	  System.out.println("Running all the test cases of infrastucture module...");
+	  for (File file : listOfTests){
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java]: Use logger for this. Do not directly print it. Is this printing necessary?

File: src/infrastructure/validation/testing/TestHarness.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  */
+  public boolean runByName(String testName){
+    String[] arrOfStr = str.split(".", 2); 
+    String testClassName = arrOfStr[0];
+    TestCase test = Class.forName(testClassName).newInstance();
+    bool result = test.run();
+    
+    //after saving the result using logger
+    return result;
+  }
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java:173]: Test harness functions do not return anything. They only save the result.

File: src/infrastructure/validation/testing/TestHarness.java [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  
+ /**
+  * Boolean method to give result of the Test Case Class
+  * @param testName    The file name of test case class including .java extension
+  * @return    <code>true</code> if the test case pass 
+  *     and actual output is same as expected output; 
+  *     <code>false</code> otherwise.
+  */
+  public boolean runByName(String testName){
+    String[] arrOfStr = str.split(".", 2);
```
Comment (Rahul-R5) [inline src/infrastructure/validation/testing/TestHarness.java]: What is 'str'? If you are trying to split test name, you need to split it as [Test name] and "Test". There won't be any punctuations. We have removed the underscore naming too.

**Review by Rahul-R5 — CHANGES_REQUESTED** [review]
Please make the required changes

File: src/infrastructure/validation/testing/TestHarness.java (hunk 5 above)
Comment (Rahul-R5) [reply src/infrastructure/validation/testing/TestHarness.java:173]: This result might be needed. Ignore the previous comment

**Comment by abhisheksaran** [comment]
RunAll() and runByName() are done.

**Comment by Rahul-R5** [comment]
Outdated

**Pull request closed without merging**